internal/controller/bucket/setup.go

```diff
@@ -45,20 +43,15 @@ var (
 func Setup(mgr ctrl.Manager, o controller.Options, c *Connector) error {
 	name := managed.ControllerName(v1alpha1.BucketGroupKind)
 
-	cps := []managed.ConnectionPublisher{managed.NewAPISecretPublisher(mgr.GetClient(), mgr.GetScheme())}
-	if o.Features.Enabled(features.EnableAlphaExternalSecretStores) {
-		cps = append(cps, connection.NewDetailsManager(mgr.GetClient(), apisv1alpha1.StoreConfigGroupVersionKind))
-	}
-
 	opts := []managed.ReconcilerOption{
 		managed.WithCriticalAnnotationUpdater(managed.NewRetryingCriticalAnnotationUpdater(mgr.GetClient())),
 		managed.WithTimeout(c.operationTimeout + time.Second),
 		managed.WithPollInterval(c.pollInterval),
-		managed.WithExternalConnecter(c),
+		managed.WithExternalConnector(c),
 		managed.WithLogger(o.Logger.WithValues("bucket reconciler", name)),
 		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
-		managed.WithConnectionPublishers(cps...),
 		managed.WithCreationGracePeriod(c.creationGracePeriod),
+		managed.WithDeterministicExternalName(true),
```

Comment:
Q: Why are this change and removing ConnectionPublisher necessary?

Reply:
Similar to above, this [PR](https://github.com/crossplane/crossplane-runtime/pull/841) allows a controller to designate a managed resource as having a deterministic name - this means crossplane will essentially "opt-out" of the previosuly problematic "pending annotation" behaviour (more details in the PR).

As for the ConnectionPublisher stuff - that has been removed upstream